common/persistence/elasticsearch/es_visibility_store.go

```diff
@@ -604,6 +643,59 @@ func getSQLFromListRequest(request *p.ListWorkflowExecutionsByQueryRequest) stri
 	return sql
 }
 
+func getSQLFromListAllRequest(request *p.InternalListAllWorkflowExecutionsByTypeRequest) string {
+	sql := "select * from dummy "
+	var earliestTimeStr, latestTimeStr, timeRange, searchString, statusFilter, whereClause, orderByClause string
+
+	if !request.EarliestTime.IsZero() && !request.LatestTime.IsZero() {
+		earliestTimeStr = strconv.FormatInt(request.EarliestTime.UnixNano()-oneMicroSecondInNano, 10)
+		latestTimeStr = strconv.FormatInt(request.LatestTime.UnixNano()+oneMicroSecondInNano, 10)
+		timeRange = fmt.Sprintf("%s between %s and %s", es.StartTime, earliestTimeStr, latestTimeStr)
+		whereClause = addToWhereClause(whereClause, timeRange)
+	}
+
+	for _, status := range request.StatusFilter {
+		if statusFilter == "" {
+			statusFilter = fmt.Sprintf("%s = %s",
+				es.CloseStatus, status)
+		}
+		statusFilter += fmt.Sprintf("or %s = %s",
+			es.CloseStatus, status)
```

Comment:
From the comments I think we are missing tests for status filters.

Reply:
updated